autofs-5.0.5 - fix ampersand escape in auto.smb

From: Ian Kent <raven@themaw.net>

CIFS share names containing an ampersand need to be escaped.
---

 CHANGELOG        |    1 +
 samples/auto.smb |    1 +
 2 files changed, 2 insertions(+)


--- autofs-5.0.5.orig/CHANGELOG
+++ autofs-5.0.5/CHANGELOG
@@ -24,6 +24,7 @@
 - fix reconnect get base dn.
 - add missing sasl mutex callbacks.
 - fix get query dn failure.
+- fix ampersand escape in auto.smb.
 
 03/09/2009 autofs-5.0.5
 -----------------------
--- autofs-5.0.5.orig/samples/auto.smb
+++ autofs-5.0.5/samples/auto.smb
@@ -26,6 +26,7 @@ $SMBCLIENT -gNL $key 2>/dev/null| awk -v
 		  # Enclose mount dir and location in quotes
 		  # Double quote "$" in location as it is special
 		  gsub(/\$$/, "\\$", loc);
+		  gsub(/\&/,"\\\\&",loc)
 		  print " \\\n\t \"/" dir "\"", "\"://" key "/" loc "\""
 		}
 	END 	{ if (!first) print "\n"; else exit 1 }
